nominations/views.py

```diff
@@ -325,22 +361,40 @@ def candidate_login(request):
 
     return render(request, 'candidate/login.html', {'form': form})
 
+
 def reset_questionnaire(request):
     app_id = request.GET.get('id')
     user = request.session['profile']
     user_id = user['user_id']
+    default_next_url = '/groups/nominations/questionnaire/edit?id=' + app_id
+
+    # if next query string exists, redirect there after
+    next_url = request.GET.get(
+        'next',
+        default_next_url
+    )
 
     try:
-        questionnaire = Application.objects.all().filter(user_id=user_id,pk=app_id).first().questionnaire
-    except (Application.DoesNotExist, KeyError):
-        # TODO: Fix the error thrown when no nomination
-        messages.error(self.request, "We could not find your questionnaire. Please try again.")
-        return redirect("/groups/nominations/dashboard?c=1")
+        application = Application.objects.all().filter(
+            user_id=user_id,
+            pk=app_id
+        ).first()
+        questionnaire = application.questionnaire
+    except (Application.DoesNotExist, AttributeError):
+        messages.error(
+            request,
+            QUESTIONNAIRE_NOT_FOUND_ERROR
+        )
+        return redirect('/groups/nominations/dashboard/')
 
+    # get the application attached to the questionnaire
     questionnaire.status = 'incomplete'
+    application.authorized_email = None
     questionnaire.save()
+    application.save()
```

Comment:
app save already calls questionnaire save too? don't need to do both?

Reply:
This is needed because while questionnaire saves the application in the questionnaire save method, authorized_email changing would not be saved in that method. Thus, an applicaiton save here is needed to save authorized_email change.

Reply:
I suggest to use a solution where application is saved only once. For example could do something like this where `skip_application` is a param that will skip the auto app save:
```
    questionnaire.save(skip_application=True)
    application.authorized_email = None
    application.save()
```

Reply:
Like this, will implement.

Reply:
Done.